File: Programmers/Greedy(G)/G_기지국 설치.java
import java.util.*;

class Solution {
    public int solution(int n, int[] stations, int w) {
        int answer = 0;
        int si = 0;
        int position = 1;
        while(position <= n) {
            // 새로 설치할 것이 범위 내에 겹치는 경우
            if(si < stations.length && stations[si] - w <= position) {
                position = stations[si++] + w + 1;
            }
            else {
                position += w * 2 + 1;
                answer++;
            }            
        }
        return answer;
    }
}

File: Programmers/Greedy(G)/G_기지국 설치.cpp
#include <iostream>
#include <vector>

using namespace std;

int solution(int n, vector<int> stations, int w) {
    int answer = 0;
    int pos = 1;
    int idx = 0;
    
    while(pos <= n) {
        if(idx < stations.size() && stations[idx] - w <= pos) {
            pos = stations[idx] + w + 1;
            idx++;
            continue;
        }
        pos += 2*w + 1;
        answer++;
    }

    return answer;
}